### PR TITLE
Add cycle_finder

### PR DESCRIPTION
**Pull request by rkajitani**
I added the recipe of cycle_finder: a de novo analysis tool for tandem and interspersed repeats based on cycle-finding. Its algorithm and performance are described in the [preprint](https://www.biorxiv.org/content/10.1101/2023.07.17.549334v2.full) (doi: https://doi.org/10.1101/2023.07.17.549334). As an practical example, this tool was used to analyse repetitive genomic sequences of the hagfish (doi: https://doi.org/10.1101/2023.05.28.542657).

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by rkajitani** [comment]
@BiocondaBot please add label

File: recipes/cycle_finder/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+{% set sha256 = "1254695585e1211b279ca036ac24594100ef1d21998e9fe1739f83364fbb7942" %}
+
+
+package:
+  name: cycle_finder
+  version: {{ version }}
+
+source:
+  sha256: '{{sha256}}'
+  url: https://github.com/rkajitani/cycle_finder/archive/refs/tags/v1.0.0.tar.gz
```
Comment (bgruening) [inline recipes/cycle_finder/meta.yaml:11]: You could use here the version token, this will help our bit to auto update the recipe with every new version

**Review by bgruening — COMMENTED** [review]
Thanks, if you like you can annotate your recipe with the your doing our extra section

**Comment by rkajitani** [comment]
@bgruening Thank you for the review! As you suggested, I have updated the meta.yaml. In addition, I have fixed the home URL.

**Review by bgruening — APPROVED** [review]

**Pull request merged**